EventFilter/TotemRawToDigi/src/RawToDigiConverter.cc

```diff
@@ -0,0 +1,271 @@
+  const TotemDAQMapping &mapping, const TotemAnalysisMask &analysisMask,
+  DetSetVector<TotemRPDigi> &rpData, DetSetVector<TotemVFATStatus> &finalStatus)
+{
+  // map to link TotemFramePosition to the conversion status
+  map<TotemFramePosition, TotemVFATStatus> status;
+
+  // common processing - frame validation
+  RunCommon(input, mapping, status);
+
+  // second loop over data
+  for (VFATFrameCollection::Iterator fr(&input); !fr.IsEnd(); fr.Next())
+  {
+    auto mappingIter = mapping.VFATMapping.find(fr.Position());
+    if (mappingIter == mapping.VFATMapping.end())
+      continue;
+  
+    // check whether the data come from RP VFATs
+    if (mappingIter->second.symbolicID.subSystem != TotemSymbID::RP)
+      throw cms::Exception("RawToDigiConverter::Run") << "VFAT is not from RP. subSystem = " <<
+        mappingIter->second.symbolicID.subSystem << endl;
+
+    // silently ignore RP CC VFATs
+    if (mappingIter->second.type != TotemVFATInfo::data)
+      continue;
+
+    // calculate ids
+    unsigned short chipId = mappingIter->second.symbolicID.symbolicID;
+    det_id_type detId = TotemRPDetId::decToRawId(chipId / 10);
+    uint8_t chipPosition = chipId % 10;
+
+    // update chipPosition in status
+    TotemVFATStatus &actualStatus = status[fr.Position()];
+    actualStatus.setChipPosition(chipPosition);
+
+    // produce digi only for good frames
+    if (actualStatus.isOK())
+    {
+      // find analysis mask (needs a default=no mask, if not in present the mapping)
+      TotemVFATAnalysisMask anMa;
+      anMa.fullMask = false;
+  
+      auto analysisIter = analysisMask.analysisMask.find(mappingIter->second.symbolicID);
+      if (analysisIter != analysisMask.analysisMask.end())
+      {            
+        // if there is some information about masked channels - save it into conversionStatus
+        anMa = analysisIter->second;
+        if (anMa.fullMask)
+          actualStatus.setFullyMaskedOut();
+        else
+          actualStatus.setPartiallyMaskedOut();
+      }
+  
+      // create the digi
+      unsigned short offset = chipPosition * 128;
+      const vector<unsigned char> &activeChannels = fr.Data()->getActiveChannels();
+    
+      for (auto ch : activeChannels)
+      {
+        // skip masked channels
+        if (!anMa.fullMask && anMa.maskedChannels.find(ch) == anMa.maskedChannels.end())
+        {
+          DetSet<TotemRPDigi> &digiDetSet = rpData.find_or_insert(detId);
+          digiDetSet.push_back(TotemRPDigi(offset + ch));
+        }
+      }
+    }
+
+    // save status
+    DetSet<TotemVFATStatus> &statusDetSet = finalStatus.find_or_insert(detId);
+    statusDetSet.push_back(actualStatus);
+  }
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void RawToDigiConverter::PrintSummaries()
+{
+  if (printErrorSummary)
+  {
+    cout << "* Error summary (error signature : number of such events)" << endl;
```

Comment:
cout is not allowed in production code.
Use LogWarning or LogPrint.
Minimally, make sure that this is never executed in the default configuration.